lib/Dialect/HW/Transforms/FlattenIO.cpp

```diff
@@ -93,11 +93,22 @@ struct InstanceOpConversion : public OpConversionPattern<hw::InstanceOp> {
       }
     }
 
-    // Create the new instance...
-    Operation *targetModule = SymbolTable::lookupNearestSymbolFrom(
-        op, op.getReferencedModuleNameAttr());
+    // Get the new module return type.
+    llvm::SmallVector<Type> newResultTypes;
+    for (auto oldResultType : op.getResultTypes())
+      if (auto structType = getStructType(oldResultType))
+        for (auto t : structType.getElements())
+          newResultTypes.push_back(t.type);
+      else
+        newResultTypes.push_back(oldResultType);
+
+    // Create the new instance, with incorrect argnames and other attributes,
```

Comment:
nit: I wouldn't call this 'incorrect', maybe the comment should say that the instance is created to target the rewritten module, which will be adjusted later on.